tests/unit/amo/reducers/test_addonsByAuthors.js

```diff
@@ -0,0 +1,170 @@
+import { ADDON_TYPE_THEME } from 'core/constants';
+import reducer, {
+  OTHER_ADDONS_BY_AUTHORS_PAGE_SIZE,
+  fetchOtherAddonsByAuthors,
+  initialState,
+  loadOtherAddonsByAuthors,
+} from 'amo/reducers/addonsByAuthors';
+import { fakeAddon } from 'tests/unit/amo/helpers';
+
+
+describe(__filename, () => {
+  describe('reducer', () => {
+    it('initializes properly', () => {
+      const state = reducer(undefined, {});
+      expect(state).toEqual(initialState);
+    });
+
+    it('ignore unrelated actions', () => {
+      // Load some initial state to be sure that an unrelated action does not
+      // change it.
+      const state = reducer(undefined, loadOtherAddonsByAuthors({
+        slug: fakeAddon.slug,
+        addons: [fakeAddon],
+      }));
+      const newState = reducer(state, { type: 'UNRELATED' });
+      expect(newState).toEqual(state);
+    });
+
+    describe('loadOtherAddonsByAuthors()', () => {
+      it('allows an empty list of add-ons', () => {
+        const state = reducer(undefined, loadOtherAddonsByAuthors({
+          slug: 'addon-slug',
+          addons: [],
+        }));
+        expect(state.byAddonSlug).toEqual({
+          'addon-slug': [],
+        });
+      });
+
+      it('adds related add-ons by slug', () => {
+        const state = reducer(undefined, loadOtherAddonsByAuthors({
+          slug: 'addon-slug',
+          addons: [fakeAddon],
+        }));
+        expect(state.byAddonSlug).toEqual({
+          'addon-slug': [fakeAddon],
+        });
+      });
+
+      it('excludes the main add-on from the list of add-ons', () => {
+        const state = reducer(undefined, loadOtherAddonsByAuthors({
+          slug: fakeAddon.slug,
+          addons: [fakeAddon],
+        }));
+        expect(state.byAddonSlug).toEqual({
+          [fakeAddon.slug]: [],
+        });
+      });
+
+      it('always ensures the page size is consistent', () => {
+        const slug = 'addon-slug';
+        const state = reducer(undefined, loadOtherAddonsByAuthors({
+          slug,
+          // This is the case where there are more add-ons loaded than needed.
+          addons: Array(OTHER_ADDONS_BY_AUTHORS_PAGE_SIZE + 2).fill(fakeAddon),
+        }));
+        expect(state.byAddonSlug[slug])
+          .toHaveLength(OTHER_ADDONS_BY_AUTHORS_PAGE_SIZE);
+      });
+    });
+
+    describe('fetchOtherAddonsByAuthors()', () => {
+      it('resets the loaded add-ons', () => {
```

Comment:
Correct. I wanted to separate reducer and action creators, but now they have the same name so.. it makes sense to merge them. Thanks!